File: python/leetcode/image-smoother.py
class Solution:
    def smoothPixel(self, M, c, r):
        total_count = 0
        total_black = 0
        c_coords = [c, c - 1, c + 1]
        r_coords = [r, r - 1, r + 1]
        for c_coord in c_coords:
            for r_coord in r_coords:
                if c_coord < len(M) and c_coord >= 0 and r_coord < len(M[0]) and r_coord >= 0:
                    total_count += 1
                    total_black += M[c_coord][r_coord]
        return total_black // total_count
    
    def imageSmoother(self, M):
        """
        :type M: List[List[int]]
        :rtype: List[List[int]]
        """
        smoothed = []
        for c in range(len(M)):
            if len(smoothed) <= c:
                smoothed.append([])
            for r in range(len(M[c])):
                smoothed[c].append(self.smoothPixel(M, c, r))
        return smoothed

File: java/leetcode/image-smoother.java
class Solution {
    public int[][] imageSmoother(int[][] M) {
        int[][] smoothed = new int[M.length][M.length > 0 ? M[0].length : 0];
        for (int c = 0; c < M.length; c++) {
            for (int r = 0; r < M[c].length; r++) {
                smoothed[c][r] = smoothPixel(M, c, r);
            }
        }
        return smoothed;
    }

    private int smoothPixel(int[][] M, int c, int r) {
        int totalCount = 0;
        int totalBlack = 0;
        for (int cCoord = c - 1; cCoord <= c + 1; cCoord++) {
            for (int rCoord = r - 1; rCoord <= r + 1; rCoord++) {
                if (cCoord < M.length && cCoord >= 0 && rCoord < M[0].length && rCoord >= 0) {
                    totalCount++;
                    totalBlack += M[cCoord][rCoord];
                }
            }
        }
        return Math.round(totalBlack / totalCount);
    }
}
